src/test/kotlin/systems/ajax/malov/stockanalyzer/repository/impl/MongoStockRecordRepositoryTest.kt

```diff
@@ -87,8 +94,9 @@ class MongoStockRecordRepositoryTest : AbstractMongoIntegrationTest {
         val actual = mongoStockRecordRepository.findTopNStockSymbolsWithStockRecords(5, from, to)
 
         // THEN
-        assertEquals(expected.size, actual.size)
-        assertEquals(expected.keys, actual.keys)
+        StepVerifier.create(actual)
+            .expectNextMatches { it.size == expected.size && it.keys == expected.keys }
```

Comment:
use assert function instead of `expectNextMatches`

Reply:
if the test fails then you cannot wind out what exactly failed: 
- it.size == expected.size
OR
- it.keys == expected.keys